src/posix/platform/netif.cpp

```diff
@@ -523,8 +523,10 @@ void AddRtAttr(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, const
 
     assert(NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len) <= aMaxLen);
     OT_UNUSED_VARIABLE(aMaxLen);
-
-    rta           = (struct rtattr *)((char *)(aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-align"
+    rta           = (struct rtattr *)(((char *)aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
```

Comment:
Have you ever tried reinterpret_cast<>?
```suggestion
    rta           = reinterpret_cast<struct rtattr *>(reinterpret_cast<char *>(aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
```

Reply:
no, i didn't try, but as mention x893 (https://github.com/openthread/openthread/issues/5647#issuecomment-708471882), - there could be broken align with reinterpret_cast.

Anyway, in previous issue (#5647) they applied pragma's error suppression

Reply:
Pretty check is failing. Can you help update?
```
--- src/posix/platform/netif.cpp	2021-12-20 02:17:29.036508320 +0000
+++ -	2021-12-20 02:18:12.622407540 +0000
@@ -525,7 +525,7 @@
     OT_UNUSED_VARIABLE(aMaxLen);
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wcast-align"
-    rta           = (struct rtattr *)(((char *)aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
+    rta = (struct rtattr *)(((char *)aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
 #pragma GCC diagnostic pop
     rta->rta_type = aType;
     rta->rta_len  = len;
 *** ERROR:  src/posix/platform/netif.cpp is not pretty.
```

Reply:
@intx82 , can you help resolve the pretty check failure?